src/mon/MDSMonitor.cc

```diff
@@ -1274,11 +1286,7 @@ int MDSMonitor::parse_role(
     mds_role_t *role,
     std::ostream &ss)
 {
-  const FSMap *relevant_fsmap = &fsmap;
-  if (mon->is_leader()) {
-    relevant_fsmap = &pending_fsmap;
-  }
-  return relevant_fsmap->parse_role(role_str, role, ss);
+  return get_working_fsmap().parse_role(role_str, role, ss);
```

Comment:
Here, just as well, we are allowing that the caller (`filesystem_command()`) end up returning to the user based on potentially uncommitted state.